org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMDocTypeFormatter.java

```diff
@@ -196,7 +196,7 @@ private void replaceLeftSpacesWith(int to, String replacement, List<TextEdit> ed
 	}
 
 	private void replaceLeftSpacesWithOneSpace(int offset, List<TextEdit> edits) {
-		formatterDocument.replaceLeftSpacesWithOneSpace(offset, edits);
+		formatterDocument.replaceLeftSpacesWithOneSpace(-1, offset, edits);

```

Comment:
Does this signal something special to XMLFormatterDocumentNew? I'm unclear on what this is supposed to do

Reply:
Yes you can see the case here (it sets the left limit to -1) : https://github.com/eclipse/lemminx/blob/07134c571566ac04f3cbc500b870e8bc7ad17dc1/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/XMLFormatterDocumentNew.java#L375

Reply:
I'm trying to remove this -1 by setting the correct offset. I started something but it is not finished.

Reply:
Ok now I set the proper leftLimit.